src/js/reducers/datasetsReducer.js

```diff
@@ -33,8 +33,23 @@ function datasetsReducer(state, action) {
       field: action.field,
       order: action.order
     }));
+  }
+
+  if (action.type === ACTIONS.FILTER_DATASET) {
+    state = setIn(state, action.id + '._filter',
```

Comment:
If we store the filter expression under `._filter` (and formula under `._formula`), the user can only specify a single filter and formula expression per dataset. Instead, perhaps we want to have more general `addTransform` and `updateTransform` actions? 

Both of these would take a `dsId` and a `def`, which is the Vega definition for a particular transform (e.g., `{"type": "filter", "test": ""}`). The reducer would then append the `def` to a dataset's `transform` array. `updateTransform` would also have to take an additional `idx` to identify which element in the `transform` array is being updated. 

This structure would also simplify the exporter logic as we don't need to do anything special to export the transforms. They'd be exported directly as part of a dataset.